lms/djangoapps/learner_dashboard/programs.py

```diff
@@ -71,13 +75,90 @@ class ProgramDetailsFragmentView(EdxFragmentView):
     Render the program details fragment.
     """
 
+    @staticmethod
+    def get_program_discussion_configuration(program_uuid):
+        return ProgramDiscussionsConfiguration.objects.filter(
+            program_uuid=program_uuid
+        ).first()
+
+    @staticmethod
+    def _get_resource_link_id(program_uuid, request) -> str:
+        site = get_current_site(request)
+        return f'{site.domain}-{program_uuid}'
+
+    @staticmethod
+    def _get_result_sourcedid(context_id, resource_link_id, user_id) -> str:
+        return f'{context_id}:{resource_link_id}:{user_id}'
+
+    def _get_lti_embed_code(self, program_discussions_configuration, request) -> str:
+        """
+        Returns the LTI embed code for embedding in the program discussions tab
+        Args:
+            program_discussions_configuration (ProgramDiscussionsConfiguration): ProgramDiscussionsConfiguration object.
+            request (HttpRequest): Request object for view in which LTI will be embedded.
+        Returns:
+            HTML code to embed LTI in program page.
+        """
+        program_uuid = program_discussions_configuration.program_uuid
+        lti_consumer = program_discussions_configuration.lti_configuration.get_lti_consumer()
+        user_id = str(request.user.id)
+        context_id = program_uuid
+        resource_link_id = self._get_resource_link_id(program_uuid, request)
+        roles = 'student'
```

Comment:
will this be static always? what about a role for staff?

Reply:
Not implemented yet. WIP